AngAcceptance.c

```diff
@@ -0,0 +1,217 @@
+
+    TTree* THTree = (TTree*)File->Get("THTree");
+    int THPMT, DHID;
+    double THTime;
+    THTree->SetBranchAddress("DHID", &DHID);
+    THTree->SetBranchAddress("THPMT", &THPMT);
+    THTree->SetBranchAddress("THTime", &THTime);
+
+    TTree* GeoTree = (TTree*)File->Get("GeoTree");
+    double PMTOri[2014][3], PMTPos[2014][3];
+    GeoTree->SetBranchAddress("PMTOri", PMTOri);
+    GeoTree->SetBranchAddress("PMTPos", PMTPos);
+    GeoTree->GetEntry(0);
+
+    //Set up histogram
+    TH1D* fitQunDH = new TH1D("fitQunDH","Angular acceptance against cosine of the reconstructed incident angle;cosine;angular acceptance",10,0,1);
+    TH1D* fitQunInAngle = new TH1D("fitQunInAngle","Cosine of the fitQun incident angle;cosine;Number of hit",10,0,1);
+    TH1D* WCSimTH = new TH1D("WCSimTH","Angular acceptance against cosine of true incident angle;cosine;angular acceptance",10,0,1);
+    TH1D* TrueInAngle = new TH1D("TrueInAngle","Cosine of the true incident angle;cosine;Number of true hits",10,0,1);
+    TH1D* fitQunTH = new TH1D("fitQunTH","Angular acceptance against cosine of true incident angle;cosine;angular acceptance",10,0,1);
+
+    TH1D* TrueBinHist[10];
+    TH1D* fitQunBinHist[10];
+    for (int i=0; i<10; i++) {
+        TrueBinHist[i] = new TH1D(("True"+to_string(i)).c_str()," ",450,0,9000);
+        fitQunBinHist[i] = new TH1D(("fitQun"+to_string(i)).c_str()," ",450,0,9000);
+    }
+
+    int TotalEventNum = EventTree->GetEntries();
+    std::cout << "Number of events: " << TotalEventNum <<
+        "\nNumber of digit hits in all events: " << DHTree->GetEntries() << "\n";
+
+    int DHTreeStartPos = 0, DHTreeEndPos, THTreeStartPos = 0, THTreeEndPos;
+    int TimeConstructionFailNum = 0, ignored_track = 0;
+    for (int i=0; i<TotalEventNum; i++) {
+        EventTree->GetEntry(i);
+        std::cout << "\r" << i << std::flush;  //Enable this line to show the progress in runtime
+        if (selection(fitQunEntrance, fitQunExit)) {
+            DHTreeStartPos += DHNum;
+            THTreeStartPos += THNum;
+            ignored_track++;
+            continue;
+        }
+
+        double fitQunEntranceTime;
+        if ((fitQunPos[2]<max_z)&&((fitQunPos[0]*fitQunPos[0]+fitQunPos[1]*fitQunPos[1])<(max_r*max_r))) {
+            fitQunEntranceTime = fitQunVerTime -
+                (1e9)*sqrt((fitQunEntrance[0]-fitQunPos[0])*(fitQunEntrance[0]-fitQunPos[0]) +
+                (fitQunEntrance[1]-fitQunPos[1])*(fitQunEntrance[1]-fitQunPos[1]) +
+                (fitQunEntrance[2]-fitQunPos[2])*(fitQunEntrance[2]-fitQunPos[2]))/TMath::Ccgs();
+        }
+        else {
+            fitQunEntranceTime = fitQunVerTime +
+                (1e9)*sqrt((fitQunEntrance[0]-fitQunPos[0])*(fitQunEntrance[0]-fitQunPos[0]) +
+                (fitQunEntrance[1]-fitQunPos[1])*(fitQunEntrance[1]-fitQunPos[1]) +
+                (fitQunEntrance[2]-fitQunPos[2])*(fitQunEntrance[2]-fitQunPos[2]))/TMath::Ccgs();
+        }
+        if (fitQunEntranceTime<0) {
```

Comment:
Why is this necessary? In real data we don't have an absolute t=0 as a reference point

Reply:
The time of fitQun vertex should be agreed with t=0 at the start of the simulation and I am constructing the track of the emitted photon stating form the entrance point. Therefore I add this for a better hit time calculation later.

Reply:
This is only true for simulation, but we don't know the proper timing for real data. If we want events with better timing resolution, we should look at other cuts but not on the absolute time which is only known for simulation